tests/introspection/tests.py

```diff
@@ -189,6 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
+    @skipUnlessDBFeature('can_introspect_index_type')
+    def test_get_constraints_indexes_types(self):
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
+        for key, val in constraints.items():
+            if val['index'] and not (val['primary_key'] or val['unique']):
+                self.assertIn('type', val)
+                self.assertEqual('btree', val['type'])
```

Comment:
Reverse ordering of args. I'm not convinced that all backends should return 'btree' -- should it return the index suffix instead? Anyway, it be changed later, do we have a ticket to resolve that issue?